apps/website/versioned_docs/version-v3.x/technical-references/zk-snark-circuits/processMessages.md

```diff
@@ -0,0 +1,107 @@
+---
+title: MACI Process Messages Circuit
+description: Introduction to the core zk-SNARK circuits of MACI
+sidebar_label: Process Messages Circuit
+sidebar_position: 3
+---
+
+[**Repo link**](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/core)
+
+This circuit allows the coordinator to prove that they have correctly processed each message in reverse order, in a consecutive batch of 5 ^ msgBatchDepth messages to the respective state leaf within the state tree. Coordinators would use this circuit to prove correct execution at the end of each Poll.
+
+The [`processMessages`](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/core/qv/processMessages.circom) circuit will try to decrypt the messages, and based on the content of the message, update within itself the trees, to generate a proof that the coordinator's off-chain processing was done correctly. In other words, the circuit takes a final state, an initial state, and the leaves (messages and user signups) - it processes these messages via the different state transitions to finally check that the expected state is correct.
+The pre-requisites for this circuit are:
+
+- the related Poll has ended
+- the state tree has been merged
+- the message tree has been merged
+
+This circuit requires the coordinator's private key, hence a proof for this circuit can only be generated by the coordinator. The private key is needed in order to generate the ECDH shared key used to decrypt the messages.
+
+![ProcessMessages](/img/circuits/processMessages_2_0.svg)
+
+:::info
+A version working with non quadratic voting (non-qv) is also [available](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/core/non-qv/processMessages.circom). This version is called `processMessagesNonQV` and is to be used when the Poll is not using the quadratic voting feature. Note that by default MACI works with quadratic voting.
+:::
+
+#### Parameters
+
+| #   | Parameter                | Description                                          |
+| --- | ------------------------ | ---------------------------------------------------- |
+| 0   | State tree depth         | Allows $(2^{n})$ signups.                            |
+| 1   | Message tree depth       | Allows $(5^{n})$ votes or key-change messages.       |
```

Comment:
we now have only 3 params, messageTreeDepth and messageBatchTreeDepth has been replace with messageBatchSize